src/message/connection-endpoint.js

```diff
@@ -293,14 +294,15 @@ ConnectionEndpoint.prototype._processInvalidAuth = function( authError, authData
  * @param   {SocketWrapper} socketWrapper
  * @param   {String} authError     String or null if auth succesfull
  * @param   {String} username
+ * @param   {Any} data     Optional data to return to the client
  *
  * @private
- * 
+ *
  * @returns {void}
  */
-ConnectionEndpoint.prototype._processAuthResult = function( authData, socketWrapper, authError, username ) {
+ConnectionEndpoint.prototype._processAuthResult = function( authData, socketWrapper, authError, username, data ) {
 	if( authError === null ) {
-		this._registerAuthenticatedSocket( socketWrapper, username );
+		this._registerAuthenticatedSocket( socketWrapper, username, ( Array.isArray( data ) ) ? data : [ data ] );
```

Comment:
I think this may be incorrect. The following is the algo for creating messages. You always want your arguments to be a single part of the message.

So if you run:

`getMsg( 'A', 'A', [ 1, 2, 3 ] );`

you'll get:

> A|A|1|2|3+

instead of;

> A|A|[1,2,3]+

So you should probably be able to run it just using:

`this._registerAuthenticatedSocket( socketWrapper, username, [ data ] );`

getMsgAlgo:
```
exports.getMsg = function( topic, action, data ) {
	var sendData = [ topic, action ],
		i;
	if( data ) {
		for( i = 0; i < data.length; i++ ) {
			if( typeof data[ i ] === 'object' ) {
				sendData.push( JSON.stringify( data[ i ] ) );
			} else {
				sendData.push( data[ i ] );
			}
		}
	}
	return sendData.join( SEP ) + C.MESSAGE_SEPERATOR;
};
```